### PR TITLE
KSP-68-69-79 User Auth + Event Logic

### PR DESCRIPTION
**Pull request by navarrojandg**
If this is pulled into main, should be able to sign up, login, logout, create event, and view event details. I think we should be able to solely focus on the availability stacking now that these are out of the way.

### CODE REVIEW

**Review by jesc0bar — CHANGES_REQUESTED** [review]
Registering a user does not route. Left comment in code for a change

**Comment by jesc0bar** [comment]
Other than the route issue I think everything look great! There are a few additions which will require future features like restricting access to database, events, etc. but overall a really great step forward. Thanks

File: src/components/RegisterForm/RegisterForm.tsx [hunk 1]
```diff
@@ -3,20 +3,41 @@ import Form from 'react-bootstrap/Form';
...
-        // eslint-disable-next-line
-        console.log('USER_REGISTER', formValue);
+        const formValue: RegisterFormData = Object.fromEntries(
+            formData.entries()
+            // eslint-disable-next-line
+        ) as any;
+        registerUser(formValue)
+            .then(({ data }) => {
+                // eslint-disable-next-line
+                console.log(`User successfully created!`, data);
```
[the remark this answers is not in the document]
Comment (navarrojandg) [reply src/components/RegisterForm/RegisterForm.tsx:33]: Yes you are correct lol i thought i had it but nope

**Comment by navarrojandg** [comment]
Okay added firebase rules, added redirect after user registers, also added redirect to event page after event creation. Overall the firebase rules are as follows:
- /users/ : only user can view their user record, and they can update their user record except for modifying their email or uid
- /events/ : anyone can create, only host can update and both host and guests are able to view event details

Let me know if i missed anything for now

**Review by jesc0bar — COMMENTED** [review]
Starting up the emulators will be a problem because functions is running with node v14 and react project is on node v16. If we aren't running emulator then it's ok. New changes look good though.

**Pull request merged**